src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1033,6 +1138,23 @@ public static Collection<FunctionMetadata> ReadFunctionMetadata(ScriptHostConfig
             // determine the script type based on the primary script file extension
             functionMetadata.ScriptType = ParseScriptType(functionMetadata.ScriptFile);
 
+            JToken isDirect;
+            if (functionConfig.TryGetValue("configurationSource", StringComparison.OrdinalIgnoreCase, out isDirect))
+            {
+                var value = isDirect.ToString().ToLowerInvariant();
+                if (value == "attributes")
+                {
+                    functionMetadata.IsDirect = true;
```

Comment:
why not do this step in TryParseFunctionMetadata where other metadata is loaded from the json?